### PR TITLE
refactor: modify codebase to use db-subgraph plugin

### PR DESCRIPTION
**Pull request by fernandomg**
# Description:

This is the first iteration towards making adding support for particular functionalities easy via a plug-in system.

In this case, it's quite convoluted as it requires quite a lot of setup, but the idea is there. No subgraphs by default, but you can add them programmatically. Yay! (https://github.com/bootnodedev/db-subgraph.git)

> [!NOTE]
> This needs to be replicated in the main dAppBooster project, but the idea is to see it working here before moving to the other steps.
>
> That's why this is a draft, but needs to be reviewed 🙏 

# Steps:

Check the db-subgraph readme and try to use it as it is here.

## Type of change:

- [x] New feature
- [ ] Bug fix
- [x] Breaking change
- [x] Enhancement
- [x] Refactoring
- [x] Chore

# How Has This Been Tested?

- [x] Manual testing
- [ ] Automated tests
- [ ] Other (explain)

# Remember to check that:

- Your code follows the style guidelines of this project
- You have performed a self-review of your code
- You have commented your code in hard-to-understand areas
- You have made corresponding changes to the documentation
- Your changes generate no new warnings

# Screenshots

(Add screenshots or videos to help test this pull request)

### CODE REVIEW

**Review by nicosampler — APPROVED** [review]
Lovely ❤️

File: src/components/pageComponents/home/Examples/demos/SubgraphDemo/index.tsx [hunk 1]
```diff
@@ -1,30 +1,287 @@
...
+import ArbitrumDefault from '@/src/components/pageComponents/home/Examples/demos/assets/Arbitrum'
+import BaseDefault from '@/src/components/pageComponents/home/Examples/demos/assets/Base'
+import OptimismDefault from '@/src/components/pageComponents/home/Examples/demos/assets/Optimism'
+import PolygonDefault from '@/src/components/pageComponents/home/Examples/demos/assets/Polygon'
+import { env } from '@/src/env'
+import { allAaveReservesQueryDocument } from '@/src/subgraphs/queries/aave/reserves'
+import { allUniswapPoolsQueryDocument } from '@/src/subgraphs/queries/uniswap/pools'
+import { withSuspenseAndRetry } from '@/src/utils/suspenseWrapper'
+
+const chainNameMapping: { [key: number]: string } = {
```
Comment (paterson1) [inline src/components/pageComponents/home/Examples/demos/SubgraphDemo/index.tsx:29]:
```suggestion
const chainNameMapping: Record<Chain['id'], string> = {
```

Not tested 🥸

File: src/components/pageComponents/home/Examples/demos/SubgraphDemo/index.tsx [hunk 2]
```diff
@@ -1,30 +1,287 @@
...
+        </Row>
+      ))}
+    </Group>
+  )
+})
+
+const uniswapNetworks = [optimism, polygon, arbitrum]
+
+const List = ({ ...restProps }) => {
+  const [currentChain, setCurrentChain] = useState<Chain | undefined>(uniswapNetworks[0])
```
Comment (paterson1) [inline src/components/pageComponents/home/Examples/demos/SubgraphDemo/index.tsx:258]: Is undefined required here if it is initialized with the first value of `uniswapNetworks`

File: src/components/pageComponents/home/Examples/demos/SubgraphDemo/index.tsx [hunk 3]
```diff
@@ -1,30 +1,287 @@
...
+import request from 'graphql-request'
+import { toast } from 'react-hot-toast'
+import { arbitrum, base, type Chain, optimism, polygon } from 'viem/chains'
+
+import { OptionsButton } from '@/src/components/pageComponents/home/Examples/demos/OptionsButton'
+import { OptionsDropdown } from '@/src/components/pageComponents/home/Examples/demos/OptionsDropdown'
+import ArbitrumDefault from '@/src/components/pageComponents/home/Examples/demos/assets/Arbitrum'
+import BaseDefault from '@/src/components/pageComponents/home/Examples/demos/assets/Base'
+import OptimismDefault from '@/src/components/pageComponents/home/Examples/demos/assets/Optimism'
+import PolygonDefault from '@/src/components/pageComponents/home/Examples/demos/assets/Polygon'
```
Comment (paterson1) [inline src/components/pageComponents/home/Examples/demos/SubgraphDemo/index.tsx:23]: At first sight, I wasn't sure if they were icons or what. import { ...., ... } from `/Icons/index.ts`. Anyway severity -1

File: src/components/pageComponents/home/Examples/demos/SubgraphDemo/index.tsx [hunk 4]
```diff
@@ -1,30 +1,287 @@
...
+      position: 'top-center',
+      id: 'copy-to-clipboard',
+    })
+  }
+
+  return <CopyButton onClick={handleCopy} value={value} />
+}
+
+// eslint-disable-next-line react-refresh/only-export-components
+export const getNetworkIcon = (chainName: string) => (
```
Comment (paterson1) [inline src/components/pageComponents/home/Examples/demos/SubgraphDemo/index.tsx:164]: `:Chain['???']`  or use switch/match with a default icon

**Review by paterson1 — APPROVED** [review]
Great work!

**Comment by fernandomg** [comment]
this was a preview draft PR... it served it's purpose

**Pull request closed without merging**